units/commit_queue.go

```diff
@@ -173,46 +193,69 @@ func (j *commitQueueJob) Run(ctx context.Context) {
 			j.AddError(sendCommitQueueGithubStatus(j.env, pr, message.GithubStateFailure, "can't get modules", ""))
 			j.dequeue(cq, nextItem)
 		} else {
-			j.AddError(errors.Wrap(cq.SetProcessing(false), "can't set processing to false"))
+			j.logError(cq.SetProcessing(false), "can't set processing to false", nextItem)
 		}
 		return
 	}
 	patchDoc.Patches = append(patchDoc.Patches, modulePatches...)
 
-	v, err := makeVersion(ctx, githubToken, projectConfig, patchDoc)
+	// populate tasks/variants matching the commitqueue alias
+	projectConfig.BuildProjectTVPairs(patchDoc, patchDoc.Alias)
```

Comment:
It surprises me that this is sufficient to add the new builds and tasks to the db. Later, `model.FinalizePatch` seems wrong, since the patch has already been finalized. Instead I would think you would call `model.AddNewTasks` or `model.AddNewBuilds`. If I'm missing something, could you explain how this approach works?

Reply:
I don't see how the patch is already finalized. 
In what sense has the patch already been finalized?

Reply:
Discussed in person.